ibis-server/app/model/metadata.py

```diff
@@ -0,0 +1,357 @@
+from app.model.data_source import DataSource
+from app.model.dto import PostgresConnectionUrl, PostgresConnectionInfo
+from enum import StrEnum, auto
+from json import loads
+from app.model.metadata_dto import (
+    CompactTable,
+    CompactColumn,
+    Constraint,
+    WrenEngineColumnType,
+    ConstraintType,
+)
+
+
+class Metadata(StrEnum):
+    postgres = auto()
+    bigquery = auto()
+
+    def get_table_list(self, connection_info):
+        if self == Metadata.postgres:
+            return self.get_postgres_table_list_sql(connection_info)
+        if self == Metadata.bigquery:
+            return self.get_bigquery_table_list_sql(connection_info)
+        raise NotImplementedError(f"Unsupported data source: {self}")
+
+    def get_constraints(self, connection_info):
+        if self == Metadata.postgres:
+            return self.get_postgres_table_constraints(connection_info)
+        if self == Metadata.bigquery:
+            return self.get_bigquery_table_constraints(connection_info)
+        raise NotImplementedError(f"Unsupported data source: {self}")
+
+    @staticmethod
+    def get_postgres_table_list_sql(
+        connection_info: PostgresConnectionUrl | PostgresConnectionInfo,
+    ):
+        sql = """
+            SELECT
+                t.table_catalog,
+                t.table_schema,
+                t.table_name,
+                c.column_name,
+                c.data_type,
+                c.is_nullable,
+                c.ordinal_position
+            FROM
+                information_schema.tables t
+            JOIN
+                information_schema.columns c ON t.table_schema = c.table_schema AND t.table_name = c.table_name
+            WHERE
+                t.table_type in ('BASE TABLE', 'VIEW')
+                and t.table_schema not in ('information_schema', 'pg_catalog')
+            """
+        res = to_json(
+            DataSource.postgres.get_connection(connection_info)
+            .sql(sql, dialect="trino")
```

Comment:
The dialect is for the sql if not the data source sql. You can use `dialect` to let ibis know your sql is what dialect to convert the sql to data source sql.
Here, you clearly use Postgres SQL. You don't provide dialect to ibis.